### DISCUSSION TITLE
what is execution_id? when getting a workflow instance from databases, we select record by instance_id and execution_id as filters, why execution id is necessary?  Is instance id not unique enough ?

### DISCUSSION THREAD

**Discussion by abserari** [cschleiden/go-workflows#189] [Q&A] [answered]
I read the source code and think that execution_id is the same life cycle as the instance id that doesn't change at all, so I don't understand why the execution_id is necessary.

**Comment by cschleiden** [comment, marked as answer]
You are right, currently it's not used at all. This relates to the Durable Task concept of "Eternal orchestrations": https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-eternal-orchestrations?tabs=csharp 

I thought of integrating this in go-workflows, too, but for now decided against it. It could be removed I just haven't gotten to it yet.

**Reply by cschleiden** [reply]
To get back to this, the need for `ContinueAsNew` and eternal orchestrations arose and I had to bring `execution_id` back, but at least this time it's being used.
